File: mns-reactor/src/test/java/cz/fb/manaus/reactor/categorizer/AllCategorizerTest.java
package cz.fb.manaus.reactor.categorizer;

import cz.fb.manaus.core.MarketCategories;
import cz.fb.manaus.core.category.BetCoverage;
import cz.fb.manaus.core.model.Side;
import cz.fb.manaus.core.test.AbstractLocalTestCase;
import cz.fb.manaus.core.test.CoreTestFactory;
import org.hamcrest.CoreMatchers;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Set;

import static org.junit.Assert.assertThat;

public class AllCategorizerTest extends AbstractLocalTestCase {

    @Autowired
    private AllCategorizer categorizer;

    @Test
    public void testCategory() throws Exception {
        assertThat(categorizer.getCategories(CoreTestFactory.newSettledBet(2d, Side.LAY), BetCoverage.EMPTY),
                CoreMatchers.is(Set.of(MarketCategories.ALL)));
    }

}

File: mns-reactor/src/test/java/cz/fb/manaus/reactor/categorizer/AllCategorizerTest.kt
package cz.fb.manaus.reactor.categorizer

import cz.fb.manaus.core.MarketCategories
import cz.fb.manaus.core.model.realizedBet
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class AllCategorizerTest {

    @Test
    fun category() {
        assertEquals(setOf(MarketCategories.ALL), AllCategorizer.getCategories(realizedBet))
    }

}